driver/k8s/logicalvolume_service.go

```diff
@@ -108,25 +109,54 @@ func NewLogicalVolumeService(mgr manager.Manager) (*LogicalVolumeService, error)
 }
 
 // CreateVolume creates volume
-func (s *LogicalVolumeService) CreateVolume(ctx context.Context, node, dc, name string, requestGb int64) (string, error) {
-	logger.Info("k8s.CreateVolume called", "name", name, "node", node, "size_gb", requestGb)
+func (s *LogicalVolumeService) CreateVolume(ctx context.Context, node, dc, name, parentID string, requestGb int64, sourceVol *topolvmv1.LogicalVolume, volumeMode, fsType string) (string, error) {
```

Comment:
Could you please make 3 functions like below and use common check logic?
- CreateVolume
- CreateVolumeWithDatasouce
- CreateSnapshot

Reply:
Sure! Created a new function that handles the common logic.
Since now, we have only 1 extra parameter (`parentID`) in the CreateVolume request, I believe it will be better to use the same function. Let me know what you think?
We now have 2 functions `CreateVolume` and `CreateSnapshot` functions with shared logic in a helper function.